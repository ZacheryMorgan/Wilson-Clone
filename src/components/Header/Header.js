import React from "react";
import "../../styles/Header.scss";

const Header = () => {
  return (
    <header className="bg-[#101820] flex items-center justify-center h-9">
      <ol className="list-disc header-list">
        <li className="underline cursor-pointer text-white flex items-center text-xs tracking-wider">
          Free Shipping on Orders Over $50
        </li>
      </ol>
    </header>
  );
};
export default Header;
