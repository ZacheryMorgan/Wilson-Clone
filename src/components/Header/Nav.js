import React from "react";
import { Link } from "react-router-dom";
import { FiUser } from "react-icons/fi";
import { BiSearch, BiShoppingBag } from "react-icons/bi";
import { GiHamburgerMenu, GiNautilusShell } from "react-icons/gi";
import "../../styles/Nav.scss";

const Nav = ({ onClick }) => {
  return (
    <nav className="border-b">
      <div className="nav-wrapper flex justify-between items-center h-12 px-5 max-w-screen-2xl mx-auto xl:h-20">
        <Link to="/">
          <GiNautilusShell className="w-8 h-8 mr-10 xl:w-12 xl:h-12 min-w-min  hover:cursor-pointer" />
        </Link>
        <ul className="site-links flex gap-14">
          <Link to="/shop">
            <li className="text-l font-bold cursor-pointer relative p-2 uppercase hidden xl:block">
              Shells
            </li>
          </Link>
          <li className="text-l font-bold cursor-pointer relative p-2 uppercase hidden xl:block">
            Equipment
          </li>
          <li className="text-l font-bold cursor-pointer relative p-2 uppercase hidden xl:block">
            Custom
          </li>
        </ul>
        <ul className="flex gap-1 nav-links">
          <button className="">
            <BiSearch className="hover:fill-red-600 cursor-pointer transition-all duration-300 w-6 h-6" />
          </button>
          <button className="hidden xl:block">
            <FiUser className="hover:stroke-red-600 cursor-pointer transition-all duration-300 w-6 h-6" />
          </button>
          <button className="" onClick={onClick}>
            <BiShoppingBag className="hover:fill-red-600 cursor-pointer transition-all duration-300 w-6 h-6 " />
          </button>
          <button className="xl:hidden">
            <GiHamburgerMenu className="hover:fill-red-600 cursor-pointer transition-all duration-300 w-6 h-6 active:border-black after:border-2" />
          </button>
        </ul>
      </div>
    </nav>
  );
};
export default Nav;
