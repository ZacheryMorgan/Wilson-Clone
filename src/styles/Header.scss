.header-list {
  list-style: none;
  counter-reset: my-awesome-counter;

  li {
    counter-increment: my-awesome-counter;
  }

  li::before {
    content: "";
    background: white;
    width: 0.6rem;
    height: 0.6rem;
    border-radius: 50%;
    display: inline-block;
    line-height: 2rem;
    color: white;
    text-align: center;
    margin-right: 0.5rem;
  }
}
