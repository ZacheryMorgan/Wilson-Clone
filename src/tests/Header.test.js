import React from 'react';
import { render } from '@testing-library/react';
import Header from '../components/Header';

describe('Header component', () => {
  it('renders Header component correctly', () => {
    const { container } = render(<Header />)
    expect(container).toMatchSnapshot()
  })
})