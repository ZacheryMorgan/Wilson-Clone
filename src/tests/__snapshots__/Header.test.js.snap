// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Header component correctly 1`] = `
<div>
  <header
    class="bg-[#101820] flex items-center justify-center h-9"
  >
    <ol
      class="list-disc header-list"
    >
      <li
        class="underline cursor-pointer text-white flex items-center text-xs tracking-wider"
      >
        Free Shipping on Orders Over $50
      </li>
    </ol>
  </header>
</div>
`;
